src/ci/scripts/should-skip-this.sh

```diff
@@ -6,15 +6,19 @@
 set -euo pipefail
 IFS=$'\n\t'
 
+git fetch "https://github.com/$GITHUB_REPOSITORY" "$GITHUB_BASE_REF"
+BASE_COMMIT="$(git merge-base FETCH_HEAD HEAD)"
```

Comment:
Not 100% sure how well I followed the preceding discussion but took a pass at addressing this with ea6ec114bfeb5b5b8afe9d2112f86df62151a336 as an additional commit

The tools job was skipped which I guess that's expected/desired given that single commit?

Please lmk if that should instead be squashed on Joshua's previous commit. Otherwise if that's fine then someone else would need to let bors know as i don't have rights

Reply:
I would not expect the tools job to be skipped. If anything in the PR touches those files, I would expect it to retry.  I can poke around, as I don't see an obvious reason why it isn't triggering.

Reply:
OK, I see the problem.  Because the script sets `pipefail`, and `grep -q` returns immediately after a match, `git` continues to emit output to a broken pipe which causes SIGPIPE, which causes the test to fail.

I think the solution is to remove `--quiet` and add `1> /dev/null 2>&1` after the `grep` command.

Reply:
Oh, looks like git has its own `--quiet` flag that could be used.  I'm not certain, but I think something like this could also work (and is faster):

```diff
diff --git a/src/ci/scripts/should-skip-this.sh b/src/ci/scripts/should-skip-this.sh
index 7ce3ec18dfb..897d38440eb 100755
--- a/src/ci/scripts/should-skip-this.sh
+++ b/src/ci/scripts/should-skip-this.sh
@@ -37,7 +37,7 @@ if git diff "$BASE_COMMIT" | grep --quiet "^index .* 160000"; then
     # Submodules pseudo-files inside git have the 160000 permissions, so when
     # those files are present in the diff a submodule was updated.
     echo "Executing the job since submodules are updated"
-elif git diff --name-only "$BASE_COMMIT" | grep --quiet src/tools/'\(clippy\|rustfmt\)'; then
+elif ! git diff --quiet "$BASE_COMMIT" -- src/tools/clippy src/tools/rustfmt; then
     # There is not an easy blanket search for subtrees. For now, manually list
     # the subtrees.
     echo "Executing the job since clippy or rustfmt subtree was updated"
```

Reply:
the former seems to have sorted things but happy to switch to the latter if that's preferable